elastictransport/elastictransport.go

```diff
@@ -586,3 +600,91 @@ func (c *Client) logRoundTrip(
 	}
 	c.logger.LogRoundTrip(req, &dupRes, err, start, dur) // errcheck exclude
 }
+
+func (c *Client) isClosed() bool {
+	select {
+	case <-c.closeC:
+		return true
+	default:
+		return false
+	}
+}
+
+// Close marks the elastictransport.Client as closed. After which it is no longer possible to perform requests.
+// It will wait until graceful shutdown of the client or the context.Context deadline is reached.
+// Close should only be called once, otherwise it will return ErrAlreadyClosed.
+func (c *Client) Close(ctx context.Context) error {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	var err error = &ErrAlreadyClosed{}
+	c.closeOnce.Do(func() {
```

Comment:
If this panics we're going to get undefined behavior, could we change this in favor of an atomic instead?